homeassistant/components/xiaomi_miio/gateway.py

```diff
@@ -116,11 +117,18 @@ def connect_gateway(self):
             try:
                 miio_cloud = MiCloud(self._cloud_username, self._cloud_password)
                 if not miio_cloud.login():
-                    raise ConfigEntryAuthFailed(
-                        "Could not login to Xiaomi Miio Cloud, check the credentials"
+                    _LOGGER.error(
+                        "Failed to login to Xiaomi Miio Cloud during setup of xiaomi"
+                        " gateway with host %s",
+                        self._host,
                     )
+                    return False
```

Comment:
When does this happen and/or should it also raise the same exception as with the invalid creds? I'm just thinking that the error message should probably give a hint to what a user can do to correct this.

Reply:
If the credentials are incorrect, it would raise an `MiCloudAccessDenied`, which is caught below and throws a `ConfigEntryAuthFailed`, triggering the re-auth flow.

IMHO in this location, we should throw a ConfigEntry instead, so Home Assistant will re-try later.

```suggestion
                    raise ConfigEntryNotReady(
                        "Failed to login to Xiaomi Miio Cloud during setup of Xiaomi"
                        " gateway with host {self._host}",
                    )
```